### PR TITLE
Fetch manifests for discovered flows

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

We didn't have access to manifests of discovered flows that have not been previously set up. This fixes it.

Before:

image

After: 

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/integrations/ha-config-integrations.ts [hunk 1]
```diff
@@ -127,6 +128,8 @@ class HaConfigIntegrations extends SubscribeMixin(LitElement) {
   @internalProperty()
   private _manifests: Record<string, IntegrationManifest> = {};
 
+  private _manifestFetched?: Set<string>;
```
Comment (spacegaier) [inline src/panels/config/integrations/ha-config-integrations.ts:131]:
```suggestion
  private _fetchedManifests?: Set<string>;
```
Might be a purely subjective thing, but that var name makes it sound like a boolean flag, whereas for actual data I would have assumed `_fetchedManifests` (plural to show it contains multiple elements and more more natural word order).

Nothing critical, so feel free to ignore the comment.

Comment (balloob) [reply src/panels/config/integrations/ha-config-integrations.ts]: Renamed it to `_extraFetchedManifests`.

Comment (spacegaier) [reply src/panels/config/integrations/ha-config-integrations.ts]: That's even better 👍

File: src/panels/config/integrations/ha-config-integrations.ts [hunk 2]
```diff
@@ -496,12 +498,33 @@ class HaConfigIntegrations extends SubscribeMixin(LitElement) {
...
+      return;
+    }
+    if (this._extraFetchedManifests) {
+      if (this._extraFetchedManifests.has(domain)) {
+        return;
+      }
+    } else {
+      this._extraFetchedManifests = new Set();
+    }
+    this._extraFetchedManifests.add(domain);
```
Comment (bramkragten) [inline src/panels/config/integrations/ha-config-integrations.ts:520]: Why do we keep track of this, to prevent double calls while we are already fetching?

**Review by bramkragten — APPROVED** [review]

File: src/panels/config/integrations/ha-config-integrations.ts (hunk 2 above)
Comment (balloob) [reply src/panels/config/integrations/ha-config-integrations.ts:520]: Yeah. If someone discovers 100 ESPHome devices, we should just stick with 1 call.

**Pull request merged**